src/Form/Widget/Generate.elm

```
@@ -352,26 +358,26 @@ isRequired wholeSchema pointer =
     isCheckboxRequired || isPropertyRequired
 
 
-fieldLabel : Maybe UI.ControlLabel -> SubSchema -> Pointer -> Maybe String
+fieldLabel : Maybe UI.ControlLabel -> SubSchema -> Pointer -> { label : Maybe String, ariaLabel : String }
 fieldLabel label schema scope =
     let
         fallback =
             schema.title
                 |> Maybe.orElse (List.last scope |> Maybe.map UI.fieldNameToTitle)
-                |> Maybe.withDefault ""
+                |> Maybe.withDefault "unreachable"
```

Aria label must always be same as the label (if such is visible) or a descriptive label (if label is not visible / not present), so it makes sense to just have label and indication if that label is hidden (if label is hidden - we can use aria-label and if not - use label).
https://github.com/scrive/elm/blob/master/libs/scrive-ui/src/UI/Input.elm#L35
https://github.com/scrive/elm/blob/master/libs/scrive-ui/src/UI/Input.elm#L53

This currently will set some aria labels to "unreachable", which doesn't sound good for accessibility...

---

Right, I misunderstood the purpose of `aria-label` a bit. I refactored the code to contain `label : String` and `hideLabel : Bool`.

The `"unreachable"` string is, in fact, never used -- it is unreachable, because `scope` is a non-empty list. Before, an empty string label (`""`) was used as a placeholder, but I thought it'd be better to make the failure more explicit.

Another fallback option would be to hide the whole widget, but that's more awkward in code and I'm not sure it's any better. I added a comment about it, do you think it's good enough? 

```haskell
-- `scope` is a non-empty list, so `fallback` is never "unreachable".
```

... Or I can revert it into an empty string again and pretend there's no issue :laughing:  :shrug:

---

The proper way would be to return a `Maybe { label : String, hideLabel : Bool }` I think, and then handle this situation whenever this function is called, but I will leave it up to you since it would require changing code that composes the control widget (you already return a maybe there, so indeed you could be not rendering the widget when an impossible state is reached).
But I think it is more issue-proof if we don't make impossible states possible with the unreachable placeholder - since there shouldn't be a widget without a label.